parking_permits_app/customer_permit.py

```diff
@@ -164,16 +172,27 @@ def update(self, data, permit_id=None):
                 }
             )
             if permit_id:
-                self.customer_permit_query.filter(id__in=permit_to_update).update(
-                    **fields_to_update
-                )
-                return self.customer_permit_query.get(id=permit_id)
+                with reversion.create_revision():
+                    self.customer_permit_query.filter(id__in=permit_to_update).update(
+                        **fields_to_update
+                    )
+                    permit = self.customer_permit_query.get(id=permit_id)
+                    event_type = EventType.CHANGED
+                    comment = get_reversion_comment(event_type, permit)
+                    reversion.set_comment(comment)
+                    return permit
 
         return self._update_fields_to_all_draft(fields_to_update)
 
     def _update_fields_to_all_draft(self, data):
-        self.customer_permit_query.filter(status=DRAFT).update(**data)
-        return self.customer_permit_query.all()
+        with reversion.create_revision():
```

Comment:
The create_revision does not track mass update, see details: https://django-reversion.readthedocs.io/en/stable/api.html#creating-revisions

We need to update the permit one by one and wrap each of them inside `create_revision` context